### DISCUSSION TITLE
How do I reference a figure sub-panel with no separation?

### DISCUSSION THREAD

**Discussion by jessicaguo** [quarto-dev/quarto-cli#7001] [Q&A]
### Description

The @fig-1 notation is super handy and easy to remember and type. However, I often refer to subpanels by letter. How can I remove the space between the figure number and the letter? 

I can do @fig-1 a, which results in a space, and if I remove the space obviously Quarto thinks I'm referring to a non-existent @fig-1a.

**Comment by cscheid** [comment]
There's no current way to do this, see https://github.com/quarto-dev/quarto-cli/issues/5481 where we're tracking the issue.

**Reply by JeffreyRStevens** [reply]
Are there any plans to revisit this? It's not quite the same as #5481, as this refers to the space after the figure number and before the subfigure letter (e.g., Figure 1a) when the subfigure is not being created in Quarto. Is there something like a `space-after-numbering` option in the works?

**Reply by cscheid** [reply]
"when the subfigure is not being created in Quarto" I don't understand what you mean.

We do support control for `space-after-numbering` for custom crossrefs: https://quarto.org/docs/authoring/cross-references-custom.html#example-supplemental-figures

You should be able to achieve what you need by switching all your crossrefs to custom crossrefs, and then render them as desired using the configuration described in the docs above.

For regular crossrefs, the way to do it depends on which format you're targeting. In LaTeX, you'll need to configure `hyperref` yourself (since that's what Quarto does). In HTML,

**Reply by JeffreyRStevens** [reply]
Thank you for your quick reply.
* By "when the subfigure is not being created in Quarto", I mean imagine that you used {patchwork} to create a figure with subfigures included and labeled in a single PNG as opposed to including and labeling separate figures in Quarto (https://quarto.org/docs/authoring/figures.html#subfigures).
* The example for custom crossrefs has a `space-before-numbering` option for something like Figure S1 (rather than Figure S 1). But when I replace `space-before-numbering` with `space-after-numbering`, I receive this error:
```
ℹ The error happened in location crossref:custom:0:space-after-numbering.
ℹ Did you mean space-before-numbering?

ERROR: Render failed due to invalid YAML.
```
* Currently, I'm rendering output in Word.

**Reply by cscheid** [reply]
Sorry, I did mean `space-before-numbering`.

**Reply by cscheid** [reply]
> * By "when the subfigure is not being created in Quarto", I mean imagine that you used {patchwork} to create a figure with subfigures included and labeled in a single PNG as opposed to including and labeling separate figures in Quarto (https://quarto.org/docs/authoring/figures.html#subfigures).

Can you try something like `@fig-1[]{.empty}a`? That is, you create an intentionally empty span instead of a space between the two parts of the thing you want to write. It's a fairly ugly hack but might just work well enough for you.

EDIT: As @pjotrhey suggests, the syntax `[@fig-1]A` is much cleaner.

**Reply by JeffreyRStevens** [reply]
That works! Many thanks, @cscheid!

**Comment by pjotrhey** [comment]
You can also just use [@fig-1]A, see https://quarto.org/docs/authoring/cross-references.html#references

**Reply by cscheid** [reply]
Thanks. You're exactly right, and I should know it, I helped write that page!

**Reply by gavinsimpson** [reply]
It would be *really* helpful to include that as a specific example on that page of the guide, and even though I read that section I didn't realise I could do this specific thing (the example has surrounding text)
